modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -3983,14 +3986,65 @@ private CompletableFuture<?> processOperationRequestWithTxRwCounter(
             }
         }
 
-        return processOperationRequest(senderId, request, isPrimary, opStartTsIfDirectRo, leaseStartTime)
-                .whenComplete((unused, throwable) -> {
-                    if (request instanceof ReadWriteReplicaRequest) {
-                        txRwOperationTracker.decrementOperationCount(
-                                rwTxActiveCatalogVersion(catalogService, (ReadWriteReplicaRequest) request)
-                        );
-                    }
-                });
+        UUID txIdLockingLwm = tryToLockLwmIfNeeded(request, opStartTsIfDirectRo);
+
+        try {
+            return processOperationRequest(senderId, request, isPrimary, opStartTsIfDirectRo, leaseStartTime)
+                    .whenComplete((unused, throwable) -> {
+                        unlockLwmIfNeeded(txIdLockingLwm, request);
+
+                        if (request instanceof ReadWriteReplicaRequest) {
+                            txRwOperationTracker.decrementOperationCount(
+                                    rwTxActiveCatalogVersion(catalogService, (ReadWriteReplicaRequest) request)
+                            );
+                        }
+                    });
+        } catch (Throwable e) {
+            try {
+                unlockLwmIfNeeded(txIdLockingLwm, request);
+            } catch (Throwable unlockProblem) {
+                e.addSuppressed(unlockProblem);
+            }
+            throw e;
+        }
+    }
+
+    private static UUID newFakeTxId() {
```

Comment:
Could you please add a javadoc or a comment that will explain why we need such method.